parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -69,6 +72,20 @@
   public static final byte[] MAGIC = "PAR1".getBytes(Charset.forName("ASCII"));
   public static final int CURRENT_VERSION = 1;
 
+  // need to supply a buffer size when setting block size. this is the default
+  // for hadoop 1 to present. copying it avoids loading DFSConfigKeys.
+  private static final int DFS_BUFFER_SIZE_DEFAULT = 4096;
+
+  // visible for testing
+  static final Set<String> BLOCK_FS_SCHEMES = new HashSet<String>();
+  static {
+    BLOCK_FS_SCHEMES.add("hdfs");
```

Comment:
HFTP is a FileSystem implementation for HDFS over HTTP. It's used to read data across clusters when they have mismatched versions of HDFS.

I guess more of what I was getting at is, there's HDFS, HFTP, and VIEWFS, and (possibly?) more. So locking in to just hdfs maybe is too strict? I wonder if there's a property of FileSystem that describes whether it supports blocks / or if there's some flag that's set (like block size) that indicates that the FS uses blocks?